lib/register.js

```diff
@@ -11,6 +11,9 @@ const toHook = []
 const proxyHandler = {
   set(target, name, value) {
     return setters.get(target)[name](value)
+  },
+  defineProperty(target, property, descriptor) {
+    return setters.get(target)[property](descriptor.value)
```

Comment:
Can we make this throw if `(!('value' in descriptor))`? The silent failure here could be disastrous.